### PR TITLE
Servicenow python

### PR DESCRIPTION
**Pull request by liorblob**

### CODE REVIEW

**Comment by liorblob** [comment]
I ran the test playbook and it passed, although it's not in the build because the instance keeps hibernating. doc: https://github.com/demisto/etc/issues/12838

File: Integrations/integration-ServiceNowPython.yml [hunk 1]
```diff
@@ -0,0 +1,1728 @@
...
+        path = 'table/' + ticket_type
+
+        return send_request(path, 'get', params = query_params)
+
+
+    def upload_file_command():
+        ticket_type = get_table_name(demisto.args().get('ticket_type', None))
+        ticket_id = demisto.args()['id']
+        file_id = demisto.args()['file_id']
+        file_name = demisto.args().get('file_name',
```
Comment (yaakovi) [inline Integrations/integration-ServiceNowPython.yml]: use `getFilePath` to get the name of the file: the function returns a dict with name, path as keys.

File: Integrations/integration-ServiceNowPython.yml [hunk 2]
```diff
@@ -0,0 +1,1728 @@
...
+                                    demisto.dt(demisto.context(), "File(val.EntryID=='"+file_id+"').Name"))
+        file_name = file_name[0] if isinstance(file_name, list) else file_name
```
Comment (yaakovi) [inline Integrations/integration-ServiceNowPython.yml]: when will it be a list?

File: Integrations/integration-ServiceNowPython.yml [hunk 3]
```diff
@@ -0,0 +1,1728 @@
...
+                return 'Cannot find computer'
+            computer = res['result'][0]
+        else:
+            computer = res['result']
+
+        if computer['u_code'] != computer_name:
+            return 'Computer not found'
+
+        hr = {
+            'Id': computer['sys_id'],
```
Comment (yaakovi) [inline Integrations/integration-ServiceNowPython.yml]: should be ID

File: Integrations/integration-ServiceNowPython.yml [hunk 4]
```diff
@@ -0,0 +1,1728 @@
...
+        if not res or 'result' not in res:
+            return 'Cannot find groups'
+
+        hr_groups = []
+        context_groups = []
+
+        for group in res['result']:
+            if group['name'] == group_name:
+                hr_groups.append({
+                    'Id': group['sys_id'],
```
Comment (yaakovi) [inline Integrations/integration-ServiceNowPython.yml]: should be ID

File: Integrations/integration-ServiceNowPython.yml [hunk 5]
```diff
@@ -0,0 +1,1728 @@
...
+            'sysparm_query': 'name=' + group_name
+        }
+
+        return send_request(path, 'get', params=query_params)
+
+    def fetch_incidents():
+        query_params = {}
+        incidents = []
+        sysparm_query = demisto.params().get('sysparm_query', None)
+        sysparm_limit = demisto.params().get('fetch_limit', DEFAULTS['limit'])
```
Comment (yaakovi) [inline Integrations/integration-ServiceNowPython.yml]: should be DEFAULTS['fetch_limit']

File: Integrations/integration-ServiceNowPython.yml [hunk 6]
```diff
@@ -0,0 +1,1728 @@
...
+        query_params['sysparm_limit'] = sysparm_limit
+        if sysparm_query:
+            query_params['sysparm_query'] += '^' + sysparm_query
+
+        path = 'table/' + ticket_type
+
+        res = send_request(path, 'get', params = query_params)
+
+        for result in res['result']:
+            incident = result
```
Comment (yaakovi) [inline Integrations/integration-ServiceNowPython.yml]: why?

File: Integrations/integration-ServiceNowPython.yml [hunk 7]
```diff
@@ -0,0 +1,1728 @@
...
+  required: false
+- display: Default ticket type of commands. Can be incident, sc_request, problem or
+    change_reuquest. Commands also take ticket_type argument
+  name: ticket_type
+  defaultvalue: incident
+  type: 0
+  required: false
+- display: ServiceNow API Version (e.g. 'v1')
+  name: api_version
+  defaultvalue: ""
```
[the remark this answers is not in the document]
Comment (liorblob) [reply Integrations/integration-ServiceNowPython.yml]: The version isn't necessary, blank can be a default value

File: Integrations/integration-ServiceNowPython.yml (hunk 2 above)
Comment (liorblob) [reply Integrations/integration-ServiceNowPython.yml]: If there's more than one file in the context

File: Integrations/integration-ServiceNowPython.yml (hunk 3 above)
Comment (liorblob) [reply Integrations/integration-ServiceNowPython.yml]: backward compatibility, can add ID in addition

File: Integrations/integration-ServiceNowPython.yml (hunk 4 above)
Comment (liorblob) [reply Integrations/integration-ServiceNowPython.yml]: backward compatibility, can add ID in addition

File: Integrations/integration-ServiceNowPython.yml (hunk 1 above)
Comment (liorblob) [reply Integrations/integration-ServiceNowPython.yml]: The file can be given as an argument, not necessarily from context

**Pull request merged**